### PR TITLE
Fix cookie redaction in logs

### LINKED ISSUE #17901: Directus user is logged out after app reload
### Describe the Bug

Everytime I reload the Directus app, the login screen appears.

### To Reproduce

Download Directus version `9.23.3`, open the admin panel, login and then refresh the page (or copy the url and open in a new tab)

### Hosting Strategy

Self-Hosted (Docker Image)

### LINKED ISSUE #17905: TypeError: cookieHeader.replace is not a function
### Describe the Bug

After updating to 9.23.3 directus started crashing with the following error:

```
TypeError: cookieHeader.replace is not a function
    at redactHeaderCookie (/directus/dist/utils/redact-header-cookies.js:7:37)
    at res (/directus/dist/logger.js:118:97)
    at Object.wrappedResSerializer [as res] (/directus/node_modules/.pnpm/pino-std-serializers@6.0.0/node_modules/pino-std-serializers/index.js:31:14)
    at Pino.asJson (/directus/node_modules/.pnpm/pino@8.11.0/node_modules/pino/lib/tools.js:132:33)
    at Pino.write (/directus/node_modules/.pnpm/pino@8.11.0/node_modules/pino/lib/proto.js:205:28)
    at Pino.LOG [as info] (/directus/node_modules/.pnpm/pino@8.11.0/node_modules/pino/lib/tools.js:62:21)
    at onResFinished (/directus/node_modules/.pnpm/pino-http@8.3.3/node_modules/pino-http/logger.js:124:15)
    at ServerResponse.onResponseComplete (/directus/node_modules/.pnpm/pino-http@8.3.3/node_modules/pino-http/logger.js:173:14)
    at ServerResponse.emit (node:events:525:35)
    at onFinish (node:_http_outgoing:989:10)
```

### To Reproduce

Our setup is using microsoft oauth for authentication.

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by br41nslug**
## Description

Takes a slightly different approach to redacting cookies. This implementation should:
- account for `response.headers.set-cookie` being an Array
- not manipulate the in-flight request/response objects
- also redact all cookies

Closes #17909
Fixes #17905
Fixes #17901

## Type of Change

- [x] Bugfix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [ ] All tests are passing locally
- [ ] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

### CODE REVIEW

File: api/src/logger.ts [hunk 1]
```diff
@@ -44,6 +44,30 @@ if (env['LOG_STYLE'] !== 'raw') {
...
+				return redactHeaderCookie(value, [`access_token`, `${env['REFRESH_TOKEN_COOKIE_NAME']}`, ...ssoCookies]);
+			}
+			if (path === 'res.headers') {
+				if ('set-cookie' in value) {
+					value['set-cookie'] = toArray(value['set-cookie']).map((cookie) =>
+						redactHeaderCookie(cookie, [`access_token`, `${env['REFRESH_TOKEN_COOKIE_NAME']}`, ...ssoCookies])
+					);
+				}
+				return value;
+			}
```
Comment (licitdev) [inline api/src/logger.ts:66]: Let's extract `cookieNames`.

```suggestion
			const path = pathParts.join('.');
			const cookieNames = [`access_token`, `${env['REFRESH_TOKEN_COOKIE_NAME']}`, ...ssoCookies];
			if (path === 'req.headers.cookie') {
				return redactHeaderCookie(value, cookieNames);
			}
			if (path === 'res.headers') {
				if ('set-cookie' in value) {
					value['set-cookie'] = toArray(value['set-cookie']).map((cookie) => redactHeaderCookie(cookie, cookieNames));
				}
				return value;
			}
```

File: api/src/logger.ts [hunk 2]
```diff
@@ -44,6 +44,30 @@ if (env['LOG_STYLE'] !== 'raw') {
 		},
 	};
 }
+if (env['LOG_STYLE'] === 'raw') {
```
Comment (licitdev) [inline api/src/logger.ts:47]: Should this be an `else` or perhaps inverting the above `!== 'raw'` block then adding the `else`?

**Review by licitdev — DISMISSED** [review]

File: api/src/logger.ts (hunk 2 above)
Comment (br41nslug) [reply api/src/logger.ts:47]: Considered that but found this "explicit if" more readable.

File: api/src/logger.ts (hunk 1 above)
Comment (br41nslug) [reply api/src/logger.ts:66]: Good one but lets pull the cookieNames list to the root so it only gets constructed once.

**Review by rijkvanzanten — APPROVED** [review]

**Review by azrikahar — APPROVED** [review]
I've attempted to add unit test to cover these redaction, and in the process I noticed `redactHeaderCookie` is using `--redacted--` but logger in general uses `--redact--`, so I've made them consistently `--redact--` 👍

PR LGTM! If the new test is okay, then we should be good to go.

File: api/src/utils/redact-header-cookies.ts [hunk 3]
```diff
@@ -1,7 +1,9 @@
+import { redactText } from '../logger';
```
Comment (paescuj) [inline api/src/utils/redact-header-cookies.ts:1]: Oh, another circular dependency 😄
Could we put that in `constants.ts` instead?

**Review by paescuj — APPROVED** [review]

**Review by rijkvanzanten — APPROVED** [review]

**Review by rijkvanzanten — APPROVED** [review]

File: api/src/utils/redact-header-cookies.ts (hunk 3 above)
Comment (azrikahar) [reply api/src/utils/redact-header-cookies.ts]: updated!

**Review by paescuj — APPROVED** [review]
LGTM, many thanks 🚀

**Pull request merged**